### PR TITLE
Update prometheus to support otel-cpp 1.11.0 and bazel 6

### PR DESCRIPTION
**Pull request by owent**
Update prometheus to support otel-cpp 1.9.1 and bazel 6

### CODE REVIEW

**Comment by owent** [comment]
@esigo @lalitb @marcalff @ThomsonTan Could you please help to review this PR?

File: .github/workflows/prometheus.yml [hunk 1]
```diff
@@ -53,25 +53,44 @@ jobs:
...
     steps:
       - name: checkout otel contrib prometheus
         uses: actions/checkout@v3
         with:
           path: "otel_cpp_contrib"
+      - name: checkout otel cpp
+        uses: actions/checkout@v3
+        with:
+          repository: "open-telemetry/opentelemetry-cpp"
+          ref: "v1.9.1"
```
Comment (esigo) [inline .github/workflows/prometheus.yml:68]: v1.11.0 is out
```suggestion
          ref: "v1.11.0"
```

File: .github/workflows/prometheus.yml [hunk 2]
```diff
@@ -84,7 +103,7 @@ jobs:
         uses: actions/checkout@v3
         with:
           repository: "open-telemetry/opentelemetry-cpp"
-          ref: "v1.6.1"
+          ref: "v1.9.1"
```
Comment (esigo) [inline .github/workflows/prometheus.yml:106]:
```suggestion
          ref: "v1.11.0"
```

File: exporters/prometheus/repository.bzl [hunk 3]
```diff
@@ -19,21 +19,35 @@ def io_opentelemetry_cpp_contrib_deps():
     maybe(
         http_archive,
         name = "io_opentelemetry_cpp",
-        sha256 = "1fc371be049b3220b8b9571c8b713f03e9a84f3c5684363f64ccc814638391a5",
-        strip_prefix = "opentelemetry-cpp-1.6.1",
+        sha256 = "668de24f81c8d36d75092ad9dcb02a97cd41473adbe72485ece05e336db48249",
+        strip_prefix = "opentelemetry-cpp-1.9.1",
```
Comment (esigo) [inline exporters/prometheus/repository.bzl:23]: v1.11.0 is released

File: exporters/prometheus/CMakeLists.txt [hunk 4]
```diff
@@ -16,7 +16,7 @@ cmake_minimum_required(VERSION 3.12)
 
 project(
   opentelemetry-cpp-contrib
-  VERSION "1.6.1" # opentelemetry-cpp-contrib 1.6.1
+  VERSION "1.9.1" # opentelemetry-cpp-contrib 1.9.1
```
Comment (esigo) [inline exporters/prometheus/CMakeLists.txt:19]: v1.11.0 is released

**Review by esigo — APPROVED** [review]
LGTM
thanks :)
sorry for the delay in review. the new version should be v1.11.0 now.

**Comment by ThomsonTan** [comment]
Thanks @owent for this PR. It looks good to me too. Can we upgrade the version of opentelemtry-cpp to v1.11.0?

**Review by ThomsonTan — APPROVED** [review]

**Review by lalitb — APPROVED** [review]
LGTM.  Thanks.

File: exporters/prometheus/src/push_exporter.cc [hunk 5]
```diff
@@ -1,23 +1,123 @@
...
+   * @return the current metricsToCollect collection
+   */
+  std::vector<::prometheus::MetricFamily> &GetCollection() { return metrics_to_collect_; }
+
+  /**
+   * Gets the maximum size of the collection.
+   *
+   * @return max collection size
+   */
+  std::size_t GetMaxCollectionSize() const noexcept { return max_collection_size_; }
```
Comment (lalitb) [inline exporters/prometheus/src/push_exporter.cc:87]: Nit - we should guard GetMaxCollectionSize ad GetCollection too ?

**Comment by owent** [comment]
> Thanks @owent for this PR. It looks good to me too. Can we upgrade the version of opentelemtry-cpp to v1.11.0?

I will update in another PR later.

File: exporters/prometheus/src/push_exporter.cc (hunk 5 above)
Comment (owent) [reply exporters/prometheus/src/push_exporter.cc:87]: Maybe we can use atomic here. Can I  change it in another PR?

File: exporters/prometheus/CMakeLists.txt (hunk 4 above)
Comment (owent) [reply exporters/prometheus/CMakeLists.txt]: Updated.

File: exporters/prometheus/src/push_exporter.cc (hunk 5 above)
Comment (owent) [reply exporters/prometheus/src/push_exporter.cc]: Fixed

**Comment by owent** [comment]
> Thanks @owent for this PR. It looks good to me too. Can we upgrade the version of opentelemtry-cpp to v1.11.0?

Done

**Pull request merged**